File: components/runtimes/nodejs26/stub/sdk.js
/**
 * Stub SDK for local development and IDE autocomplete.
 * At runtime, the real sdk module bundled with the nodejs26 container is used instead.
 *
 * Install: npm install @kyma-project/sdk
 */

/** @returns {import('express').Request} */
function getRequest() {
    throw new Error('sdk stub: not available outside a running Kyma function');
}

/** @returns {import('express').Response} */
function getResponse() {
    throw new Error('sdk stub: not available outside a running Kyma function');
}

/** @returns {import('@opentelemetry/api').Tracer} */
function getTracer() {
    throw new Error('sdk stub: not available outside a running Kyma function');
}

/**
 * @param {string} type
 * @param {string} source
 * @param {*} data
 * @param {object} [optionalAttributes]
 */
function emitCloudEvent(type, source, data, optionalAttributes) {
    throw new Error('sdk stub: not available outside a running Kyma function');
}

module.exports = { getRequest, getResponse, getTracer, emitCloudEvent };

File: components/runtimes/python314/stub/sdk.py
"""
Stub SDK for local development and IDE autocomplete.
At runtime, the real sdk module bundled with the python314 container is used instead.

Install: pip install kyma-sdk
"""
from flask import Request
from opentelemetry import trace


def get_request() -> Request:
    """Returns the current Flask request object."""
    raise RuntimeError("sdk stub: not available outside a running Kyma function")


def get_tracer() -> trace.Tracer:
    """Returns the configured OpenTelemetry tracer."""
    raise RuntimeError("sdk stub: not available outside a running Kyma function")


def emit_cloud_event(event_type: str, source: str, data, optional_attributes: dict = None):
    """Publishes a CloudEvent to the configured publisher proxy."""
    raise RuntimeError("sdk stub: not available outside a running Kyma function")
